src/v2/git_url.rs

```diff
@@ -71,6 +71,34 @@ impl GitUrl {
             }
         }
     }
+
+    /// Extract the repository part of the URL
+    pub fn repository(&self) -> String {
+        lazy_static! {
+            static ref REPO_PARSE: Regex = Regex::new(r#"([^#]*)"#).unwrap();
+        }
+        REPO_PARSE.captures(&self.url).unwrap().at(1).unwrap().to_string()
+    }
+
+    /// Extract the optional branch part of the git URL
+    pub fn branch(&self) -> Option<String> {
+        lazy_static! {
+            static ref BRANCH_PARSE: Regex = Regex::new(r#".*#([^:]+)"#).unwrap();
```

Comment:
Now that I think about it, I'm uneasy about having two different regular expressions that try to parse the URL. It would be better to have a single function that handles all URL parsing, probably with a single regex. Also, the right hand side should probably be anchored with `$`.